third_party/blink/renderer/modules/cobalt/h5vcc_experiments/h_5_vcc_experiments.cc

```diff
@@ -0,0 +1,166 @@
+  auto promise = resolver->Promise();
+
+  EnsureReceiverIsBound();
+
+  base::Value::Dict experiment_config_dict;
+
+  base::Value::Dict features;
+  for (auto& feature_name_and_value : experiment_configuration->features()) {
+    features.Set(feature_name_and_value.first.Utf8(),
+                 feature_name_and_value.second);
+  }
+  experiment_config_dict.Set(cobalt::kExperimentConfigFeatures,
+                             std::move(features));
+
+  base::Value::Dict feature_params;
+  std::string param_value;
+  for (auto& param_name_and_value : experiment_configuration->featureParams()) {
+    if (param_name_and_value.second->IsString()) {
+      param_value = param_name_and_value.second->GetAsString().Utf8();
+    } else if (param_name_and_value.second->IsLong()) {
+      param_value = std::to_string(param_name_and_value.second->GetAsLong());
+    } else if (param_name_and_value.second->GetAsBoolean()) {
+      param_value = "true";
+    } else if (!param_name_and_value.second->GetAsBoolean()) {
+      param_value = "false";
+    } else {
+      resolver->Reject();
+    }
+    feature_params.Set(param_name_and_value.first.Utf8(), param_value);
+  }
+  experiment_config_dict.Set(cobalt::kExperimentConfigFeatureParams,
+                             std::move(feature_params));
+
+  base::Value::List experiment_ids;
+  for (int exp_id : experiment_configuration->experimentIds()) {
+    experiment_ids.Append(exp_id);
+  }
+  experiment_config_dict.Set(cobalt::kExperimentConfigExpIds,
+                             std::move(experiment_ids));
+
+  remote_h5vcc_experiments_->SetExperimentState(
+      std::move(experiment_config_dict),
+      WTF::BindOnce(&H5vccExperiments::OnSetExperimentState,
+                    WrapPersistent(this), WrapPersistent(resolver)));
+
+  return promise;
+}
+
+void H5vccExperiments::OnSetExperimentState(ScriptPromiseResolver* resolver) {
+  resolver->Resolve();
+}
+
+ScriptPromise H5vccExperiments::resetExperimentState(
+    ScriptState* script_state,
+    ExceptionState& exception_state) {
+  auto* resolver = MakeGarbageCollected<ScriptPromiseResolver>(
+      script_state, exception_state.GetContext());
+
+  EnsureReceiverIsBound();
+
+  remote_h5vcc_experiments_->ResetExperimentState(
+      WTF::BindOnce(&H5vccExperiments::OnResetExperimentState,
+                    WrapPersistent(this), WrapPersistent(resolver)));
+
+  return resolver->Promise();
+}
+
+void H5vccExperiments::OnResetExperimentState(ScriptPromiseResolver* resolver) {
+  resolver->Resolve();
+}
+
+WTF::Vector<uint32_t> H5vccExperiments::activeExperimentIds() {
+  EnsureReceiverIsBound();
+  if (!has_experiment_ids_) {
+    remote_h5vcc_experiments_->GetActiveExperimentIds(&active_experiment_ids_);
+    has_experiment_ids_ = true;
+  }
+  return active_experiment_ids_;
+}
+
+String H5vccExperiments::getFeature(const String& feature_name) {
+  EnsureReceiverIsBound();
+  h5vcc_experiments::mojom::blink::OverrideState result;
```

Comment:
nit: `result` makes me think of a boolean success/fail. Can we name
this sth like `platform_override_state` or `platform_state` ?